### PR TITLE
Remove the outdated vineyard-cli doc and improve python API References.

### LINKED ISSUE #1579: Remove the outdated vineyard-cli documentation
Describe your problem
---------------------

Remove the outdataed doc https://v6d.io/notes/references/ctl.html.

### PR DESCRIPTION
**Pull request by dashanji**
What do these changes do?
-------------------------

- Remove the outdated vineyard-cli doc.
- Add the `vineyard.csi.read/write` API and `python -m vineyard.ctl` API to the python API References.
- Bump up sphinx to v7.2.3 to address the error `AttributeError: 'PosixPath' object has no attribute 'rstrip'`.

Related issue number
--------------------

Fixes #1579

### CODE REVIEW

File: .github/build-docs.sh [hunk 1]
```diff
@@ -6,13 +6,14 @@ set -o pipefail
 
 # python dependencies
 pip3 install -U "Pygments>=2.4.1"
+pip3 install -U "sphinx>=7.2.3"
```
Comment (sighingnow) [inline .github/build-docs.sh:9]: Why the version constraint in `requirements-dev.txt` doesn't work?

File: docs/notes/references/python-api.rst [hunk 2]
```diff
@@ -126,3 +126,21 @@ Streams
 
 .. autoclass:: vineyard.io.recordbatch.RecordBatchStream
     :members:
+
+CSI Driver
+----------
```
Comment (sighingnow) [inline docs/notes/references/python-api.rst:131]: `Interacting with the CSI driver`

File: .github/build-docs.sh (hunk 1 above)
Comment (sighingnow) [reply .github/build-docs.sh:9]: Keep those version constraints in `requirements-dev.txt`.

**Review by sighingnow — CHANGES_REQUESTED** [review]

File: .github/build-docs.sh (hunk 1 above)
Comment (dashanji) [reply .github/build-docs.sh:9]: > Why the version constraint in `requirements-dev.txt` doesn't work?

Get the following error in my local machine while building the docs.

```bash
Exception occurred:
  File "/usr/local/lib/python3.10/dist-packages/breathe/project.py", line 116, in __init__
    self._default_build_dir = os.path.dirname(app.doctreedir.rstrip(os.sep))
AttributeError: 'PosixPath' object has no attribute 'rstrip'
```

Related issue:
https://github.com/sphinx-doc/sphinx/issues/11605

File: docs/notes/references/python-api.rst (hunk 2 above)
Comment (dashanji) [reply docs/notes/references/python-api.rst]: Done.

File: .github/workflows/build-test.yml [hunk 3]
```diff
@@ -145,7 +145,7 @@ jobs:
 
           # install python packages for codegen, and io adaptors
           sudo pip3 install -U "Pygments>=2.4.1"
-          sudo pip3 install -r requirements-setup.txt -r requirements.txt -r requirements-dev.txt
+          sudo pip3 install -U -r requirements-setup.txt -r requirements.txt -r requirements-dev.txt
```
Comment (sighingnow) [inline .github/workflows/build-test.yml:148]: No `-U` here please.

**Pull request merged**